tests/testDipoleFitter.py

```diff
@@ -0,0 +1,252 @@
+        """Initialize the parameters."""
+        np.random.seed(666)
+        self.display = False
+        self.verbose = False
+        self.w, self.h = 100, 100  # size of image
+
+        self.xc = xc  # xcenters of two dipoles in image
+        self.yc = yc  # ycenters of two dipoles
+        self.flux = np.array([2500., 2345.])  # fluxes of pos/neg lobes
+        self.gradientParams = np.array([10., 3., 5.])
+
+        self.offsets = np.array([-2., 2.])  # pixel coord offsets between lobes of dipoles
+        self.testImage = ipUtils.DipoleTestImage(
+            xcenPos=self.xc + self.offsets,
+            ycenPos=self.yc + self.offsets,
+            xcenNeg=self.xc - self.offsets,
+            ycenNeg=self.yc - self.offsets,
+            flux=self.flux, fluxNeg=self.flux,
+            gradientParams=self.gradientParams)
+
+        self.catalog = self.testImage.detectDipoleSources()
+
+
+# First, test the algorithm itself (fitDipole()):
+# Create a simulated diffim (with dipoles) and a linear background gradient in the pre-sub images
+#   then compare the input fluxes/centroids with the fitted results.
+class DipoleFitTest(lsst.utils.tests.TestCase):
+    """A test case for dipole fit algorithm/plugin/task.
+
+    Test the dipole fitting algorithm on two dipoles within an image with
+    simulated noise.
+    """
+
+    def testDipoleAlgorithm(self):
+        """Test the dipole fitting algorithm. Test that the resulting
+        fluxes/centroids are very close to the input values for both
+        dipoles in the image.
+        """
+        params = DipoleFitTestGlobalParams()
+
+        if params.verbose:
+            for s in params.catalog:
+                fp = s.getFootprint()
+                print(fp.getBBox(), fp.getNpix())
+                for pk in fp.getPeaks():
+                    print('FOOTPRINT CENTER:', pk.getIy(), pk.getIx(), pk.getPeakValue())
+
+        offsets = params.offsets
+        testImage = params.testImage
+        for i, s in enumerate(params.catalog):
+            alg = DipoleFitAlgorithm(testImage.diffim, testImage.posImage, testImage.negImage)
+            result = alg.fitDipole(
+                s, rel_weight=1., separateNegParams=False,
+                verbose=params.verbose, display=params.display)
+
+            self.assertClose((result.psfFitPosFlux + abs(result.psfFitNegFlux))/2.,
+                             params.flux[i], rtol=0.02)
+            self.assertClose(result.psfFitPosCentroidX, params.xc[i] + offsets[i], rtol=0.01)
+            self.assertClose(result.psfFitPosCentroidY, params.yc[i] + offsets[i], rtol=0.01)
+            self.assertClose(result.psfFitNegCentroidX, params.xc[i] - offsets[i], rtol=0.01)
+            self.assertClose(result.psfFitNegCentroidY, params.yc[i] - offsets[i], rtol=0.01)
+
+    def runDetection(self, params):
+
+        # Create the various tasks and schema -- avoid code reuse.
+        testImage = params.testImage
```

Comment:
If the only thing you're using from params is testImage, then just pass that as the argument.